### PR TITLE
feat: Support specify build target

### PR DESCRIPTION
**Pull request by VoVAllen**
Fix https://github.com/tensorchord/envd/issues/403
Support syntax like `envd up -f :build_gpu` to specify the function to be called

### CODE REVIEW

File: examples/mnist/build.envd [hunk 1]
```diff
@@ -6,7 +6,11 @@ def build():
     #config.pip_index(url = "https://pypi.tuna.tsinghua.edu.cn/simple")
     install.python_packages([
         "tensorflow",
-        "numpy",
+        # "numpy",
```
Comment (gaocegege) [inline examples/mnist/build.envd:9]: Please remove the comment

File: examples/mnist/README.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Then you can open jupyter notebook at [`http://localhost:8888`](http://localhost
 ## Quick start (GPU)
 
 ```bash
-$ envd up -f build_gpu.envd
+$ envd up -f :build_gpu # Call build_gpu function in build.envd
```
Comment (gaocegege) [inline examples/mnist/README.md:16]: We also need to add the doc for the change. Please open an issue to keep track of the doc change.

**Review by gaocegege — APPROVED** [review]
LGTM with some nits

File: examples/mnist/README.md (hunk 2 above)
Comment (VoVAllen) [reply examples/mnist/README.md:16]: at #547

File: examples/mnist/build.envd (hunk 1 above)
Comment (VoVAllen) [reply examples/mnist/build.envd]: Removed

**Comment by gaocegege** [comment]
LGTM

**Pull request merged**